tasks/shared_libraries/recovery/test/unit_tests/test_shared_recovery.py

```diff
@@ -0,0 +1,181 @@
+        #grabbing queue contents after the message is sent
+        queue_contents = queue.receive_messages()
+
+        self.assertEqual(queue_contents[0].body, json.dumps(new_data))
+
+# ---------------------------------------------------------------------------------------------------------------------------------
+
+    def test_post_status_for_job_to_queue_all_null(self):
+            """
+            Test that sending a message to SQS queue using post_status_for_job_to_queue() function returns the same expected message.
+            The optional variables are all set to None.
+            """
+            test_sqs = boto3.resource("sqs")
+            queue = test_sqs.create_queue(QueueName="uni-test-queue.fifo")
+            request_method = shared_recovery.RequestMethod.NEW
+            job_id= '1234'
+            granule_id= '6c8d0c8b-4f9a-4d87-ab7c-480b185a0250'
+            status_id = shared_recovery.OrcaStatus.PENDING
+            request_method = shared_recovery.RequestMethod.NEW
+            archive_destination = None
+            request_time = None
+            completion_time = None
+            db_queue_url = queue.url
+            #this is the expected message body that should be received
+            new_data = {"job_id": job_id, "granule_id": granule_id}
+
+            shared_recovery.post_status_for_job_to_queue(
+                job_id,granule_id,status_id,
+                request_time,completion_time,archive_destination,request_method,db_queue_url
+                )
+            #grabbing queue contents after the message is sent
+            queue_contents = queue.receive_messages()
+
+            self.assertEqual(queue_contents[0].body, json.dumps(new_data))
+
+# ---------------------------------------------------------------------------------------------------------------------------------
+
+    def test_post_status_for_job_to_queue_no_null(self):
+            """
+            Test that sending a message to SQS queue using post_status_for_job_to_queue() function returns the same expected message.
+            The optional variables are all set to non-nullable values.
+            """
+            test_sqs = boto3.resource("sqs")
+            queue = test_sqs.create_queue(QueueName="uni-test-queue.fifo")
+            request_method = shared_recovery.RequestMethod.NEW
+            job_id= '1234'
+            granule_id= '6c8d0c8b-4f9a-4d87-ab7c-480b185a0250'
+            status_id = shared_recovery.OrcaStatus.PENDING
+            request_method = shared_recovery.RequestMethod.NEW
+            archive_destination = 'archive-destination'
+            request_time = '2019-07-17T17:36:38.494918'
+            completion_time = '2019-07-18T17:36:38.494918'
+            db_queue_url = queue.url
+            #this is the expected message body that should be received
+            new_data = {"job_id": job_id, "granule_id": granule_id, "request_time": request_time, 
```

Comment:
status changed to SUCCESS.

Reply:
We may want to add a TODO item here or a card. Let's have a quick discussion with @AndrewDorn on where that business logic should live so that unit tests for that business logic can be created.